### PR TITLE
CD-353: VMWare vCenter Monitoring - System Health

### PR DESCRIPTION
**Pull request by mariemm97**

### CODE REVIEW

File: examples/http/vmware_vcenter_monitoring_system_health.js [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+            console.log(response.body);
+            d.resolve("N/A");
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+// Function to get URLs for system health information
+function getHealthSystemsUrls(url) {
```
Comment (mohamedkallel) [inline examples/http/vmware_vcenter_monitoring_system_health.js:82]: there is no need for this function, you can use directly httpGet(url, sessionId)

File: examples/http/vmware_vcenter_monitoring_system_health.js [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+        getHealthSystemsUrls("/api/appliance/health/storage")(sessionId), //Get storage health
+        getHealthSystemsUrls("/api/appliance/health/swap")(sessionId), //Get swap health
+        getHealthSystemsUrls("/api/appliance/health/database-storage")(sessionId), //Get database storage health
+        getHealthSystemsUrls("/api/appliance/health/software-packages")(sessionId) //Get information on available software updates available in the remote vSphere Update Manager repository
+    ]);
+}
+
+// This function extracts data from the response and populates the custom table 
+function extractData(data) {
+    var recordIds = [
```
Comment (mohamedkallel) [inline examples/http/vmware_vcenter_monitoring_system_health.js:104]: make this as a global configuration variable
config = {id: "system-health", url: "/api/appliance/health/system" ...}
in getSystemHealthInfo build a list of promises based on this config then in extract data make the same code based on the iteration on the id of the configuration

**Review by mohamedkallel — CHANGES_REQUESTED** [review]

**Review by mohamedkallel — APPROVED** [review]

File: examples/http/vmware_vcenter_monitoring_system_health.js [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+            console.log(response.body);
+            d.resolve("N/A");
+        } 
+        d.resolve(JSON.parse(body));
+    });
+    return d.promise;
+}
+
+var config = [
+    { id: "system-health", url: "/api/appliance/health/system" }, //Get overall health of system
```
Comment (GeorgiTsatsev) [inline examples/http/vmware_vcenter_monitoring_system_health.js:82]: the table name is "System Health" 
having -health is redundant

File: examples/http/vmware_vcenter_monitoring_system_health.js [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+    { id: "load", url: "/api/appliance/health/load" }, //Get load health
+    { id: "memory-health", url: "/api/appliance/health/mem" }, //Get memory health
```
Comment (GeorgiTsatsev) [inline examples/http/vmware_vcenter_monitoring_system_health.js:84]: having -health is redundant

File: examples/http/vmware_vcenter_monitoring_system_health.js [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    { id: "applmgmt-service-health", url: "/api/appliance/health/applmgmt" }, //Get health status of applmgmt services
```
Comment (GeorgiTsatsev) [inline examples/http/vmware_vcenter_monitoring_system_health.js:85]: having -health is redundant

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**